var/spack/repos/builtin/packages/openfoam-com/package.py

```diff
@@ -292,6 +292,11 @@ class OpenfoamCom(Package):
 
     provides('openfoam')
     depends_on('mpi')
+
+    # Newest versions requires openmpi+thread_multiple for collated output
+    conflicts('^openmpi~thread_multiple', when='@1712:')
+    conflicts('^openmpi~thread_multiple', when='@develop')
```

Comment:
you don't need this line, `1712 < develop`, see packaging guide http://spack.readthedocs.io/en/latest/packaging_guide.html#version-comparison